awx/api/serializers.py

```diff
@@ -4618,12 +4618,15 @@ def validate(self, attrs):
             object_actual = self.context['view'].get_object()
         else:
             object_actual = None
-        for field in notification_class.init_parameters:
+        for field, params in notification_class.init_parameters.items():
             if field not in attrs['notification_configuration']:
-                missing_fields.append(field)
+                if 'default' in params:
+                    attrs['notification_configuration'][field] = params['default']
+                else:
+                    missing_fields.append(field)
                 continue
```

Comment:
Do we want this `continue` to be indented into the `else` block?

Reply:
Changed!